network/src/peers/peers.rs

```diff
@@ -286,22 +286,44 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
         }
     }
 
+    ///
     /// Broadcasts a connection request to all disconnected peers.
-    async fn connect_to_disconnected_peers(&self, count: usize) {
-        trace!("Connecting to disconnected peers");
+    ///
+    async fn connect_to_disconnected_peers(&self) {
+        // Local address must be known by now.
+        let own_address = self.local_address().unwrap();
+
+        // Fetch the number of connected and connecting peers.
+        let number_of_connected_peers = self.peer_book.number_of_connected_peers() as usize;
+        let number_of_connecting_peers = self.peer_book.number_of_connecting_peers() as usize;
+        let number_of_peers = number_of_connected_peers + number_of_connecting_peers;
+
+        // Check if this node server is below the permitted number of connected peers.
+        let min_peers = self.config.minimum_number_of_connected_peers() as usize;
+        if number_of_peers >= min_peers {
+            return;
+        }
+
+        // Set the number of peers to attempt a connection to.
+        let count = min_peers - number_of_peers;
+
+        trace!("Connecting to {} disconnected peers", count);
 
         // Iterate through a selection of random peers and attempt to connect.
         let random_peers = self
             .peer_book
             .disconnected_peers()
             .iter()
             .map(|(k, _)| k)
+            .filter(|peer| **peer != own_address)
```

Comment:
this filter should also exclude the bootnodes, which we might already actively be trying to connect to

Reply:
Resolved in post networking